draft-ietf-quic-qlog-main-schema.md

```diff
@@ -1873,17 +1873,99 @@ still provide adequate output for incomplete logs.
+not logging the field(s) in question. While deletion completely protects the
+data in the deleted fields from the risk of compromise, it also reduces the
+utility of the dataset as a whole. As such, a balance should be found between
+logging these fields and the potential risks inherent in their (involuntary)
+disclosure. This balance depends on the use case at hand (e.g., research
+datasets might have different requirements to live operational troubleshooting).
+Capturing the minimal amount of data required for a specific purpose can help to
+minimize the risks associated with data usage. qlog implementations that provide
+fine-grained control over the inclusion of data fields, ideally on a
+per-use-case or per-connection basis, improve the ability to minimize data.
+
+Any data that is determined to be necessary for a use case at hand could be
+logged or captured. As per {{!RFC6973}}, operators must be aware that such data
+will be at risk of compromise. As such, measures should be taken to firstly
+reduce the risk of compromise and secondly reduce the risk of abuse of
+compromised data. While a full discussion of both aspects is out of scope for
+this document, the following paragraphs discuss high-level considerations that
+can be applied to qlog data.
+
+To reduce the risk of compromise, operators can take measures such as: limiting
+the length of time that data is stored, encrypting data in transit and at rest,
+limiting access rights to the data, and auditing data usage practices. qlog
+deployments that provide integrated options for automated or manual data
+deletion and (aggressive) aggregation, improve the ability to minimize the risk
+of compromise.
+
+To reduce the risk of data abuse after compromise, data can be anonymized,
+pseudonymized, otherwise permutated/replaced, truncated, (re-)encrypted, or
+aggregated. A partial discussion of applicable techniques (especially for IP
+address information) can be found in Appendix B of {{!DNS-PRIVACY=RFC8932}}.
+Operators should, however, be aware that many of these techniques have been
+shown to be insufficient to safeguard user privacy and/or to protect user
+identity, especially if a qlog data set is large or easily correlated against
+other data sources. Differential privacy techniques might provide mitigations to
+this risk; see (TODO: add survey paper reference).
+
+Finally, qlog operators should consider the interplay between their use case
+needs and end user rights or preferences (e.g., in the form of cookie laws, the
+GDPR, and other regional policies). While active user participation (as
+indicated by {{!RFC6973}}) on a per-qlog basis is difficult, as logs are often
+captured out-of-band to the main user interaction and intent, general user
+expectations should be taken into account. qlog deployments that provide
+mechanisms to integrate the capture, storage and removal of qlogs with more
+general, often pre-existing, user consent and privacy control systems, improve
```

Comment:
This might be a personal bugbear, but I don't think that mention of "user consent" - particularly as it is applied here - is a good idea.  The idea that a typical user of a service might consent to this form of data collection is a little fanciful, especially given the nature of the risks.

Reply:
I wasn't sure if you had doubts about this entire paragraph of just the "user consent" terminology in this one sentence. 

Since I personally feel it's important to at least mention user preference and that qlog collection should be tied to that (as should all data collection about a user), I interpreted it as just the terminology and changed to "user preference" for now. I will open a separate issue to discuss this further.